\documentclass[final,t]{beamer}
\mode<presentation>
{
%  \usetheme{Warsaw}
%  \usetheme{Aachen}
%  \usetheme{Oldi6}
%  \usetheme{I6td}
  \usetheme{I6dv}
%  \usetheme{I6pd}
%  \usetheme{I6pd2}
}
% additional settings
\setbeamerfont{itemize}{size=\normalsize}
\setbeamerfont{itemize/enumerate body}{size=\normalsize}
\setbeamerfont{itemize/enumerate subbody}{size=\normalsize}
% additional packages
\usepackage{ragged2e}
\usepackage{verbatim}
\usepackage{textcomp}
\usepackage[ampersand]{easylist}
\usepackage{lipsum}
\usepackage{times}
\usepackage{amsmath,amsthm, amssymb, latexsym}
\usepackage{exscale}
%\boldmath
\usepackage{booktabs, array}
%\usepackage{rotating} %sideways environment
\usepackage[english]{babel}
\usepackage[latin1]{inputenc}
\usepackage[orientation=landscape,size=custom,width=111.76,height=86.36,scale=1]{beamerposter}
\addtobeamertemplate{block begin}{}{\justifying}
\usepackage[export]{adjustbox}
\listfiles
\graphicspath{{figures/}}
% Display a grid to help align images
%\beamertemplategridbackground[1cm]

\title{\huge YAX: Taxonomic assignment using metagenomic shotgun sequences}
\author{Evan T. Bolyen$^{a}$, Mike R. DeBerg$^{a}$, Andrew J. Hodel$^{a}$, Hayden T. Westbrook$^{a}$, and Viacheslav Y. Fofanov$^{a}$}
\institute{$^{a}$School of Computing, Informatics, and Cybersystems - Northern Arizona Univ.}

% abbreviations
\usepackage{xspace}
\makeatletter
\DeclareRobustCommand\onedot{\futurelet\@let@token\@onedot}
\def\@onedot{\ifx\@let@token.\else.\null\fi\xspace}
\def\eg{{e.g}\onedot} \def\Eg{{E.g}\onedot}
\def\ie{{i.e}\onedot} \def\Ie{{I.e}\onedot}
\def\cf{{c.f}\onedot} \def\Cf{{C.f}\onedot}
\def\etc{{etc}\onedot}
\def\vs{{vs}\onedot}
\def\wrt{w.r.t\onedot}
\def\dof{d.o.f\onedot}
\def\etal{{et al}\onedot}
\makeatother

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{document}
\begin{frame}{}
  \begin{columns}[t]
    \begin{column}{.3\linewidth}
        \begin{alertblock}{\includegraphics[width=.9\linewidth]{assets/yak}\newline\newline}
            We introduce a novel pipeline called YAX which is used to assign taxonomy to provided samples. Whether it is a sample of dirt, subway grime, or your own gut, YAX will help to identify members of the community using high throughput metagenomic shotgun sequences.
          \newline
        \end{alertblock}
        \begin{block}{Intended Audience}
            \textbf{Features for Biologists:}
            \begin{itemize}
                \item[$\bullet$]Pipeline configuration based on textual input file
                \item[$\bullet$]Pipeline run recieves user input in text format
                \item[$\bullet$]Easy to understand error messages
                \item[$\bullet$]Parameter validation accomplished upon submission
                \item[$\bullet$]Dependency verification checks locations before run
            \end{itemize}
            \vspace{1cm}
            \textbf{Features for Bioinformatics Developers:}
            \begin{itemize}
                \item[$\bullet$]Modular design allows for pipeline reconfiguration
                \item[$\bullet$]Basic taxonomic tools included
            \end{itemize}
        \end{block}

        \begin{block}{Requirements}
            \textbf{Modular}
            \begin{itemize}
                \item[$\bullet$]Incorporate developing technologies
                \item[$\bullet$]Evaluate different processes
            \end{itemize}
            \vspace{1cm}
            \textbf{Fail Early}
            \begin{itemize}
                \item[$\bullet$]Validate parameters before running
                \item[$\bullet$]Verify dependencies before running
                \item[$\bullet$]Prevent failure in the middle of a run for spurious reasons
            \end{itemize}
            \vspace{1cm}
            \textbf{Coverage Representation}
            \begin{itemize}
                \item[$\bullet$]Provide biologists with relevant details
                \item[$\bullet$]Every identified taxon is qualified with coverage data
            \end{itemize}
            \vspace{1cm}
            \textbf{State System}
            \begin{itemize}
                \item[$\bullet$]Manage component requirements
                \item[$\bullet$]Control and record all data produced
                \item[$\bullet$]Eliminates redundant recomputation
                \item[$\bullet$]Allows parameter exploration
            \end{itemize}
        \end{block}



      %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


    \end{column}
    \begin{column}{.3\linewidth}
        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
        \begin{block}{Purpose}
            YAX is ultimately a pipeline used for taxonomic assignment of a set of sample sequences. However it is designed to
            be dynamic in a way that allows for not only parameter exploration with as little recomputation as possible but also
            for modification of the pipeline itself with as little effor as possible. This makes YAX capable of rapid recalculations
            and reconfigurations.
        \end{block}

        \begin{block}{Solution Overview}
            The first step of YAX's iterative alignment approach compares the sample set to the entire NCBI database. This is done
            with high thresholds so that it can be accomplished as quickly as possible. If a read from the sample set aligns to more than
            one taxonomic group we can also filter this information out because of its ambiguity. The focus of this step is to eliminate
            obviously wrong answers.
            \includegraphics[width=1\linewidth]{assets/Whole} \\
            After we have obtained a subset of the entire database from the first step we can loosen the alignment thresholds to allow more
            reads from the sample set to align to reference genomes. This allows us to "use the whole yak" and squeeze as much information
            out of the sample set as possible. Of course in this step we still disregard those reads that align to taxonomic groups outside our
            thresholds as ambiguous.
            \includegraphics[width=0.679\linewidth, right]{assets/Subset}
        \end{block}


    \end{column}

    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \begin{column}{.3\linewidth}
        \begin{block}{Architecture}
            \includegraphics[width=1\linewidth]{assets/arch}\newline\newline

        \end{block}
        \begin{block}{Key Features}
            The Best Taxonomic Aligner You Will See Today!
            Configuration File creation at pipeline Initialization
            \begin{itemize}
              \item[$\bullet$]Easy user interaction
              \item[$\bullet$]Parameter validation
              \item[$\bullet$]Dependency verification
              \item[$\bullet$]Persistent use records
            \end{itemize}
            Indiana Jones the intrepid artifact hunter (named after the dog)
            \begin{itemize}
              \item[$\bullet$]YAX is aware of previously produced output
              \item[$\bullet$]Artifacts are reusable to reduce recomputation
              \item[$\bullet$]Parameter exploration
              \item[$\bullet$]Rapid error recovery
            \end{itemize}
        \end{block}
        \begin{block}{Future Work}
            YAX is designed to be highly modular and capable of adapting along side technological developments. As part of its development
            YAX was built to not only provide one solution but to investigate a number of possible solutions. The near future of YAX will
            see new modules developed to make fool use of its flexibility.

        \end{block}
        \begin{block}{References}
            \begin{itemize}
              {\small
              \item[1]Stack Overflow.
              }
          \end{itemize}
        \end{block}

        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \end{column}
  \end{columns}

\end{frame}

\end{document}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% Local Variables:
%%% mode: latex
%%% TeX-PDF-mode: t
%%% End:
